### ISSUE TITLE
Use mse and sgd, the gradient explode

### ISSUE THREAD

**Issue by as754770178** [pytorch/pytorch#31016]
I calculate coordinate loss in Yolo by mse(nn.MSELoss), and train model by sgd, the gradient explode.
```
mse = nn.MSELoss(reduce=False).to(device)
ce = nn.CrossEntropyLoss(size_average=False).to(device)

loss_coord_center = coor_xy_scale * (coord_mask * mse(coord_center, tcoord_center)).sum()
loss_coord_wh = coor_hw_scale * (coord_mask * mse(coord_wh, tcoord_wh)).sum()
```

1. When I use Adam optimizer and mse, the gradient can't explode.
2. When I don't use mse(repalce by ce(nn.CrossEntropyLoss) or iou loss) and SGD optimizer, the gradient can't explode.

cc @vincentqb

**Comment by vincentqb** [comment]
This may not be a due to a bug in pytorch. I recommend opening a discussion on the forum, [here](https://discuss.pytorch.org/new), to see what would be most appropriate in your case.

If you believe this is due to a bug in pytorch, please provide an example with code so we can reproduce.
